app/Importers/BerylDataImporter.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Importers;
+
+use GuzzleHttp\Exception\GuzzleException;
+use Symfony\Component\DomCrawler\Crawler;
+
+class BerylDataImporter extends DataImporter
+{
+    private const COUNTRY_NAME = "United Kingdom";
+
+    protected Crawler $sections;
+
+    public function extract(): static
+    {
+        try {
+            $response = $this->client->get("https://beryl.cc/where-you-can-hire");
+            $html = $response->getBody()->getContents();
+        } catch (GuzzleException) {
+            $this->createImportInfoDetails("400", self::getProviderName());
+
+            $this->stopExecution = true;
+
+            return $this;
+        }
+
+        $crawler = new Crawler($html);
+        $this->sections = $crawler->filter("div > section > div .inner");
+
+        if (count($this->sections) === 0) {
+            $this->createImportInfoDetails("204", self::getProviderName());
+            $this->stopExecution = true;
+        }
+
+        return $this;
+    }
+
+    public function transform(): void
+    {
+        if ($this->stopExecution) {
+            return;
+        }
+
+        $existingCityProviders = [];
+
+        foreach ($this->sections as $section) {
+            foreach ($section->childNodes as $node) {
+                if ($node->nodeName === "div") {
+                    foreach ($node->childNodes as $city) {
+                        if ($city->nodeName === "h3") {
+                            $cityName = $city->nodeValue;
+                        }
+
+                        $eScootersFound = false;
+
+                        if ($city->nodeValue === "e-Scooters") {
+                            $eScootersFound = true;
+                        }
+
+                        if ($eScootersFound) {
+                            $cityName = str_replace(["E-scooters", "and ", ","], "", $cityName);
+                            $cityName = trim($cityName);
+
+                            if ($cityName === "Bournemouth Christchurch Poole") {
+                                $arrayofcityNames = explode(" ", $cityName);
```

Comment:
Create array with cities?

Reply:
It has to stay this way, because those cities are written in one sentence, so i have to separate them.

Reply:
I would like to rename it to camelCase.
`arrayOfCityNames`

And all other variables related to this.

Reply:
Should be fine now(hopefully)